### PR TITLE
Add additional column to CandidateEmailSendCountsExport

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

Following feedback on https://github.com/DFE-Digital/apply-for-teacher-training/pull/4120, this export would benefit from a count of how many unique candidates have received each email type.

## Changes proposed in this pull request
- Include an additional query that counts unique recipient email addresses per mail_template
- Clean up some of the naming and object access (ie—unnecessary use of hash syntax)

## Guidance to review
- Review diff
- Download the export from the review app

## Link to Trello card
https://trello.com/c/K7FdnVOg

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/support_interface/candidate_email_send_counts_export.rb [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
     end
 
   private
 
-    def emails_with_counts
+    def unique_recipients_per_email
+      distinct_mail_templates_and_recipients = Email.select('DISTINCT "to", "mail_template"').to_sql
+      Email
+        .select('mail_template, COUNT(mail_template) AS unique_recipients')
+        .from("(#{distinct_mail_templates_and_recipients}) AS template_recipient_combos GROUP BY template_recipient_combos.mail_template")
```
Comment (stevehook) [inline app/services/support_interface/candidate_email_send_counts_export.rb:21]: I think `Email.select('count(distinct "to"), mail_template').group(:mail_template)` works too. Wondering whether you could just add the `count(distinct "to")` in the main query and drop this one altogether.

File: app/services/support_interface/candidate_email_send_counts_export.rb [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+      emails_with_total_send_counts.map do |send_count_record|
+        unique_recipients = unique_recipients_hash[send_count_record.mail_template]
         {
-          'Email' => email_hash['mail_template'],
-          'Send count' => email_hash['send_count'],
-          'Last sent at' => email_hash['last_sent_at'],
+          'Email' => send_count_record.mail_template,
+          'Send count' => send_count_record.send_count,
+          'Last sent at' => send_count_record.last_sent_at,
+          'Unique recipients' => unique_recipients,
```
Comment (stevehook) [inline app/services/support_interface/candidate_email_send_counts_export.rb:10]: :+1:

**Review by stevehook — APPROVED** [review]
Looks good as is. Just had a suggestion for simplifying the query a bit - happy to take a second look if needed.

File: app/services/support_interface/candidate_email_send_counts_export.rb (hunk 1 above)
Comment (malcolmbaig) [reply app/services/support_interface/candidate_email_send_counts_export.rb]: I think you're absolutely right, thanks for the suggestion! New commit added.

**Review by stevehook — APPROVED** [review]
Tested review app - all good.

**Pull request merged**